classes/controller/Controller.php

```diff
@@ -536,9 +536,15 @@ public function removeJS($js_uri, $check_path = true)
      * @param string|null $version jQuery library version
      * @param string|null $folder jQuery file folder
      * @param bool $minifier if set tot true, a minified version will be included
+     *
+     * @deprecated 1.7.7 jQuery is always included, this method should no longer be used
      */
     public function addJquery($version = null, $folder = null, $minifier = true)
     {
+        @trigger_error(
```

Comment:
Mmm, it is always included in `AdminController` but what about front controller? Nothing is sure in this case, and it might be difficult to detect if it was added since now it is bundled in the core js (cf your other PR https://github.com/PrestaShop/PrestaShop/issues/14929)
Maybe you should modify the `FrontController::addJquery` method in the other PR by the way?

Reply:
@jolelievre  The addJquery method exists in the FrontController, but it is empty, a comment specifies that jquery is already added by the core. So this should not be an issue.

Reply:
Yes but it is added in the core.js no? What if someone uses a theme without including core.js but uses this addJquery method instead?
The real question actually is I'm wondering what's the use of this method if jquery is always added by the core (one way or another) ?

Reply:
This method is useless and should not be used anymore, if someone did not include core.js and called this method, as of today he doesn't have jquery, because the method is empty, it does nothing...

Reply:
ok, so we take upon ourselves that this method is now useless?
but this is weird then, because `addJquery` is empty in admin and front controller, but not in the parent Controller class..
Well anyway it doesn't prevent your PR from working, it just doesn't seem logic 😅
We can leave it as is and avoid a too long discussion